pkg/controller/controller_utils/controller_utils.go

```diff
@@ -0,0 +1,42 @@
+package controller_utils
+
+import (
+	"regexp"
+	"strings"
+)
+
+const (
+	metricsNameDelimiter = "_"
+	loggerNameDelimiter  = "-"
+)
+
+type ControllerNamer interface {
+	MetricsName() string
+	LoggerName() string
+}
+
+type ControllerName []string
+
+func (c ControllerName) clean(delimiter string) ControllerName {
```

Comment:
Would recommend making this a New function instead of a clean. If It's clean we need to recompute the cleaned version every time we get the MetricsName and LoggerName. That's wasted compute when we can just compute a single time in a new function then store. With a New function is also makes it so we don't have to export the controllerName type.

```Go
const (
	metricsNameDelimiter = "_"
	loggerNameDelimiter  = "-"
)

type ControllerNamer interface {
	MetricsName() string
	LoggerName() string
}

type controllerName []string

// NewControllerName takes in a slice with each word in the name as a separate string in the slice then returns a ControllerNamer
func NewControllerName(name []string) controllerName {
   cn := make(controllerName, len(name))
   for i, w := range name {
      cn[i] = strings.ToLower(w)
      cn[i] = removeSpace(c[i])
   }
   return cn
}

func removeSpace(s string) string {
	rr := make([]rune, 0, len(s))
	for _, r := range s {
		if !unicode.IsSpace(r) {
			rr = append(rr, r)
		}
	}
	return string(rr)
}

func (c controllerName) MetricsName() string {
   return strings.Join(c, metricsNameDelimiter)
}

func (c controllerName) LoggerName() string {
   return strings.Join(c, loggerNameDelimiter)
}
```